File: adapter/src/main/java/com/hello7890/adapter/DataViewModule.java
package com.hello7890.adapter;



public abstract class DataViewModule<T> extends BaseViewModule<T> {


    public void setData(T data){
        _getDataList().clear();
        _getDataList().add(data);
        notifyItemChanged(0);
    }
    public T getData() {
        return _getDataList().isEmpty()?null:_getDataList().get(0);
    }
}

File: adapter/src/main/java/com/hello7890/adapter/DataViewModule.kt
package com.hello7890.adapter

abstract class DataViewModule<T> : BaseViewModule<T>() {
    fun setData(data: T) {
        dataList.clear()
        dataList.add(data)
        notifyItemChanged(0)
    }

    val data: T?
        get() = if (dataList.isEmpty()) null else _getDataList()[0]
}